chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -627,6 +642,104 @@ impl NetworkState {
+        debug_assert!(!body.allow_sending_to_self());
+        Ok(match body {
+            RoutedMessageBody::TxStatusRequest(account_id, tx_hash) => self
+                .client
+                .send_async(TxStatusRequest { tx_hash, signer_account_id: account_id })
+                .await
+                .ok()
+                .flatten()
+                .map(|response| RoutedMessageBody::TxStatusResponse(*response)),
+            RoutedMessageBody::TxStatusResponse(tx_result) => {
+                self.client.send(TxStatusResponse(tx_result.into()));
+                None
+            }
+            RoutedMessageBody::BlockApproval(approval) => {
+                self.client.send(BlockApproval(approval, peer_id));
+                None
+            }
+            RoutedMessageBody::ForwardTx(transaction) => {
+                self.client
+                    .send_async(ProcessTxRequest {
+                        transaction,
+                        is_forwarded: true,
+                        check_only: false,
+                    })
+                    .await
+                    .ok();
+                None
+            }
+            RoutedMessageBody::PartialEncodedChunkRequest(request) => {
+                self.shards_manager_adapter.send(
+                    ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkRequest {
+                        partial_encoded_chunk_request: request,
+                        route_back: msg_hash,
+                    },
+                );
+                None
+            }
+            RoutedMessageBody::PartialEncodedChunkResponse(response) => {
+                self.shards_manager_adapter.send(
+                    ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkResponse {
+                        partial_encoded_chunk_response: response,
+                        received_time: clock.now().into(),
+                    },
+                );
+                None
+            }
+            RoutedMessageBody::VersionedPartialEncodedChunk(chunk) => {
+                self.shards_manager_adapter
+                    .send(ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunk(chunk));
+                None
+            }
+            RoutedMessageBody::PartialEncodedChunkForward(msg) => {
+                self.shards_manager_adapter
+                    .send(ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkForward(msg));
+                None
+            }
+            RoutedMessageBody::ChunkStateWitnessAck(ack) => {
+                self.partial_witness_adapter.send(ChunkStateWitnessAckMessage(ack));
+                None
+            }
+            body => {
+                tracing::error!(target: "network", "Peer received unexpected message type: {:?}", body);
+                None
+            }
+        })
+    }
+
+    pub fn receive_routed_message_maybe_to_self(&self, body: RoutedMessageBody) {
```

Comment:
I'm confused by this function. The `maybe` in the function name indicates it handles multiple possibilities but the function body seems to assume it is handling a self-message.

Reply:
Right, so I believe the nomenclature can be improved. The high level idea is that we have two types of network messages, (1) those that we handle sending to ourselves, and (2) those that we do not handle sending to ourselves.

For now, category (1) has only chunk endorsement and partial witness related messages.

Now, in the implementation, we have two functions, currently named `receive_routed_message_maybe_to_self` and `receive_routed_message`, handling messages of type (1) and (2) respectively.

receive_routed_message_maybe_to_self doesn't indicate that the message was received from us, but more so, that it's a handler for type (1) of messages, as the function is used elsewhere as well.

We could name these functions as `receive_self_routable_message` and `receive_routed_message` or something similar. Idk, very open to name suggestions.

Reply:
Ok I see now, thanks for the explanation. From what I understand we don't actually gain anything by separating these functions in this way. And it seems kind of clunky to drop the sender `peer_id` for a message which was actually received from a peer just because the message type as a whole is self-routable (although I understand that for all self-routable types which currently exist, `peer_id` arg is unused anyway).

I would propose:
- Keep a unified function `receive_routed_message` in `network_state` which handles all variants. Peer actor can simply pass all received routed messages to this function. 
- When `send_message_to_account` handles a self-routed message, it would call the same `receive_routed_message` using the local node's id as the `peer_id` arg.
- Inside `receive_routed_message`, we can check whether peer_id matches the local node's id and if it does, execute `debug_assert!(body.allow_sending_to_self());`.

Another idea would be to have separate `receive_routed_message_from_peer` and `receive_routed_message_from_self` functions. This is perhaps the simplest and clearest approach, with the downside of duplicating the handling logic for the self-routable variants.